src/main/java/com/frcteam3636/frc2024/subsystems/arm/ArmIO.kt

```diff
@@ -0,0 +1,166 @@
+
+            var absoluteEncoderPosition = Radians.zero()!!
+
+            var rightCurrent = Volts.zero()!!
+            private var leftCurrent = Volts.zero()!!
+
+            var rightVelocity = RadiansPerSecond.zero()!!
+            var leftVelocity = RadiansPerSecond.zero()!!
+
+            var absoluteEncoderConnected = false
+
+
+            override fun toLog(table: LogTable) {
+                table.put("Right Arm Motor Position", rightPosition)
+                table.put("Left Arm Motor Position", leftPosition)
+
+                table.put("Right Arm Motor Current", rightCurrent)
+                table.put("Left Arm Motor Current", leftCurrent)
+
+                table.put("Right Arm Motor Velocity", rightVelocity)
+                table.put("Left Arm Motor Velocity", leftVelocity)
+
+                table.put("Absolute Encoder Connected",absoluteEncoderConnected )
+            }
+
+            override fun fromLog(table: LogTable) {
+                rightPosition = table.get("Right Arm Motor Position", rightPosition)
+                leftPosition = table.get("Left Arm Motor Position", leftPosition)
+
+                rightCurrent = table.get("Right Arm Motor Current", rightCurrent)
+                leftCurrent = table.get("Left Arm Motor Current", leftCurrent)
+
+                rightVelocity = table.get("Right Arm Motor Velocity", rightVelocity)
+                leftVelocity = table.get("Left Arm Motor Velocity", leftVelocity)
+
+                absoluteEncoderConnected = table.get("Absolute Encoder Connected", absoluteEncoderConnected)
+            }
+        }
+
+        fun updateInputs(inputs: ArmInputs)
+
+        fun setPosition(position: Measure<Angle>)
+
+    fun setVoltage(volts: Measure<Voltage>)
+    }
+
+    class ArmIOReal: ArmIO{
+
+        private val leftMotor = TalonFX(CTREMotorControllerId.LeftArmMotor)
+
+        private val rightMotor = TalonFX(CTREMotorControllerId.RightArmMotor)
+
+        private val absoluteEncoder = DutyCycleEncoder(DigitalInput(7))
+
+        override fun updateInputs(inputs: ArmIO.ArmInputs) {
+            inputs.position = Rotations.of(absoluteEncoder.absolutePosition)
+            inputs.absoluteEncoderConnected = absoluteEncoder.isConnected
+
+            inputs.absoluteEncoderPosition = Rotations.of(absoluteEncoder.absolutePosition)
+
+            inputs.leftPosition = Rotations.of(leftMotor.position.value)
+            inputs.leftVelocity = RotationsPerSecond.of(leftMotor.velocity.value)
+
+
+            inputs.rightPosition = Rotations.of(rightMotor.position.value)
+            inputs.rightVelocity = RotationsPerSecond.of(rightMotor.position.value)
+
+            inputs.rightCurrent = Volts.of(leftMotor.motorVoltage.value)
+
+            leftMotor.setPosition(inputs.absoluteEncoderPosition.`in`(Rotations))
+            rightMotor.setPosition(inputs.absoluteEncoderPosition.`in`(Rotations))
+        }
+
+        override fun setPosition(position: Measure<Angle>) {
```

Comment:
i think it would be good to call this something else - like "pivotToPosition" - because this could be confused with setting the encoder offset